common-files/.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,87 @@
+# Contribution guide
+
+**Never report security issues on GitHub or other public channels (Gitter/Twitter/etc.), 
+follow the instruction from [Jenkins Security](https://jenkins.io/security/) to 
+report it on [Jenkins Jira](https://issues.jenkins-ci.org)**
+
+## Why should you contribute
+
+You can contribute in many ways, and whatever you choose we're grateful!
+Source code contribution is the obvious one but we also need your feedback and if you don't really want to participate 
+in the implementation directly you may still have great ideas about features we need (or should get rid of).
+
+We have our vision for the plugin and we have an experience with maintaining Jenkins instances, 
+but the plugin is not supposed to solve only our problems. Surely we haven't experienced all of them... 
+That's why we want to hear from you.
+
+Whenever you report a problem please provide information about:
+
+- Plugin version
+- Jenkins version
+- Operating system
+- Description!
+
+### Regarding source code contribution ways of working
+
+- For larger contributions create an issue for any required discussion
+- Implement solution on a branch in your fork
+- Make sure to include issue number (if created) in commit message, and make the message speak for itself
+- Once you're done create a pull request and ask at least one of the maintainers for review
+  - Remember to title your pull request properly as it is used for release notes
+
+Never push directly to this repository!
+
+## Run Locally
+
+Prerequisites: _Java_, _Maven_ & _IntelliJ IDEA_
```

Comment:
Do we really want IDE-specific instructions like this? Lots of people are likely to use VS Code, etc. (FWIW in NetBeans you just press F6; all of this is automated.)

Anyway I think detailed documentation like this does not belong here where it will not be manintained. Write up curated guides on jenkins.io and just offer a link here.